### PR TITLE
HAProxy: Add ability to customize the ingress-operator image

### PR DESCRIPTION
**Pull request by gcs278**
### Description
Adds the ability to customize the ingress-operator image and adds additional verification and logging for the router image customization.

### Fixes
I would have to do more difficult workarounds for customizing the ingress-operator image. This makes it easier.

### CODE REVIEW

**Comment by sjug** [comment]
Logging the `${HAPROXY_IMAGE}` already exists: https://github.com/cloud-bulldozer/e2e-benchmarking/blob/master/workloads/router-perf-v2/common.sh#L54-L55

**Comment by gcs278** [comment]
/hold
Just want to get a couple more tests with this.

File: workloads/router-perf-v2/README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ It's possible to tune the default configuration through environment variables. T
 | SAMPLES               | Number of samples to perform of each test | `2` |
 | HOST_NETWORK          | Enable hostNetwork in the mb client | `true` |
 | NUMBER_OF_ROUTERS     | Number of routers to test | `2` |
+| INGRESS_OPERATOR_IMAGE     | Image to use for the ingress-operator | `default image` |
```
Comment (gcs278) [inline workloads/router-perf-v2/README.md:59]: idk really what to put for the default here, it's just the default image

**Comment by gcs278** [comment]
@sjug okay I feel good about this now have been running a couple times with this. Let me know your thoughts.

**Comment by gcs278** [comment]
Actually, just found a bug, let's hold off until I get a patch set up.

**Comment by gcs278** [comment]
Okay fixed.

File: workloads/router-perf-v2/README.md (hunk 1 above)
Comment (sjug) [reply workloads/router-perf-v2/README.md:59]: Move it under `HAPROXY_IMAGE` (at the bottom) and put the default the same: "unset".

Also add the `INGRESS_OPERATOR_IMAGE` to the env.sh as I did with the `HAPROXY_IMAGE` (commented). A good public example, or something close to it, of the OPERATOR_IMAGE would be good. This way we're consistent.

**Comment by gcs278** [comment]
Actually, just found a minor edge case. Hold one second

**Review by sjug — CHANGES_REQUESTED** [review]

File: workloads/router-perf-v2/common.sh [hunk 2]
```diff
@@ -48,19 +48,57 @@ deploy_infra(){
   oc rollout status -n http-scale-client deploy/http-scale-client
 }
 
-tune_liveness_probe(){
+configure_ingress_images() {
   log "Disabling cluster version operator"
   oc scale --replicas=0 -n openshift-cluster-version deploy/cluster-version-operator
+
+  # Scale up operator, because if it's off, then patches below will have no effect
+  log "Scaling ingress-operator up to 1"
```
[the remark this answers is not in the document]
Comment (gcs278) [reply workloads/router-perf-v2/common.sh]: done

**Pull request merged**